src/main/java/thewaypointers/trafficsimulator/gui/MapPanel.java

```diff
@@ -135,8 +157,74 @@ public void paint(Graphics g){
+    }
+
+    public  void  draw_road(JunctionDTO junctionDTO,Direction direction,Graphics g,int x,int y){
+        g.setColor(ROAD_COLOR);
+        RoadDTO road=junctionDTO.getRoad(direction);
+
+        if (direction==Direction.Up){
+            int upx=x;
+            int upy=(int )(y-road.getLength());
+            g.fillRect(upx,upy,ROAD_WIDTH,(int)road.getLength());
+            Graphics2D g2 = (Graphics2D)g;
+            g2.setColor(LANE_SEPARATOR_LINE_COLOR);
+            float [] arr={15.0f,10.0f};
+            BasicStroke stroke = new BasicStroke(1,BasicStroke.CAP_BUTT, BasicStroke.JOIN_BEVEL, 1.0f,arr,0);
+            g2.setStroke(stroke);
+            Line2D.Float line = new Line2D.Float(upx+25,upy,upx+25,upy+(int)road.getLength());
+            g2.draw(line);
+            BasicStroke stroke2=new BasicStroke();
+            g2.setStroke(stroke2);
+        }
+        if (direction==Direction.Down){
+            int down_x=x;
+            int down_y=y+ROAD_WIDTH;
+            g.fillRect(down_x,down_y,ROAD_WIDTH,(int)road.getLength());
+            Graphics2D g2 = (Graphics2D)g;
+            g2.setColor(LANE_SEPARATOR_LINE_COLOR);
+            float [] arr={15.0f,10.0f};
+            BasicStroke stroke = new BasicStroke(1,BasicStroke.CAP_BUTT, BasicStroke.JOIN_BEVEL, 1.0f,arr,0);
+            g2.setStroke(stroke);
+            Line2D.Float line = new Line2D.Float(down_x+25,down_y,down_x+25,down_y+(int)road.getLength());
+            g2.draw(line);
+            BasicStroke stroke2=new BasicStroke();
+            g2.setStroke(stroke2);
+        }
+        if (direction==Direction.Right){
+            int right_x=x+ROAD_WIDTH;
+            int right_y=y;
+            g.fillRect(right_x,right_y,(int)road.getLength(),ROAD_WIDTH);
+            Graphics2D g2 = (Graphics2D)g;
+            g2.setColor(LANE_SEPARATOR_LINE_COLOR);
+            float [] arr={15.0f,10.0f};
+            BasicStroke stroke = new BasicStroke(1,BasicStroke.CAP_BUTT, BasicStroke.JOIN_BEVEL, 1.0f,arr,0);
+            g2.setStroke(stroke);
+            Line2D.Float line = new Line2D.Float(right_x,right_y+25,right_x+(int)road.getLength(),right_y+25);
+            g2.draw(line);
+            BasicStroke stroke2=new BasicStroke();
+            g2.setStroke(stroke2);
+        }
+        if (direction==Direction.Left){
+            int left_x=(int )(x-road.getLength());
+            int left_y=y;
+            g.fillRect(left_x,left_y,(int)road.getLength(),ROAD_WIDTH);
+            Graphics2D g2 = (Graphics2D)g;
+            g2.setColor(LANE_SEPARATOR_LINE_COLOR);
+            float [] arr={15.0f,10.0f};
+            BasicStroke stroke = new BasicStroke(1,BasicStroke.CAP_BUTT, BasicStroke.JOIN_BEVEL, 1.0f,arr,0);
+            g2.setStroke(stroke);
+            Line2D.Float line = new Line2D.Float(left_x,left_y+25,left_x+(int)road.getLength(),left_y+25);
+            g2.draw(line);
+            BasicStroke stroke2=new BasicStroke();
+            g2.setStroke(stroke2);
+        }
```

Comment:
Most of the content of those ifs doesn't change, only the parameters. You should change the parameters in the ifs (maybe rework it into a switch instead of a group of ifs?) and move the rest of the work outside of ifs.